### PR TITLE
CORE-14685: Apply review comments for UniquenessChecker and BackingStore.

### PR DESCRIPTION
**Pull request by chrisr3**
Review changes after #4251.

### CODE REVIEW

File: testing/uniqueness/uniqueness-utilities/src/main/kotlin/net/corda/uniqueness/utils/UniquenessAssertions.kt [hunk 1]
```diff
@@ -26,6 +26,8 @@ import java.time.Instant
  */
 @Suppress("SpreadOperator", "TooManyFunctions")
 object UniquenessAssertions {
+    private const val UNSPECIFIED_VALUE = "<UNSPECIFIED-VALUE>"
```
Comment (relyafi) [inline testing/uniqueness/uniqueness-utilities/src/main/kotlin/net/corda/uniqueness/utils/UniquenessAssertions.kt]: I think the changes to this file are no longer required.

Comment (chrisr3) [reply testing/uniqueness/uniqueness-utilities/src/main/kotlin/net/corda/uniqueness/utils/UniquenessAssertions.kt]: Perhaps not, but they're still potentially useful for future tests. Consider them a gift :-).

Comment (relyafi) [reply testing/uniqueness/uniqueness-utilities/src/main/kotlin/net/corda/uniqueness/utils/UniquenessAssertions.kt]: Actually I wouldn't do it like this, so you're probably better reverting. Also not great to have unused code, logic is trivial and can be re-added later if needed.

Comment (chrisr3) [reply testing/uniqueness/uniqueness-utilities/src/main/kotlin/net/corda/uniqueness/utils/UniquenessAssertions.kt]: Sorry, I've run out of time to work on this PR. Either merge it or just close it.

**Review by relyafi — APPROVED** [review]

**Review by blsemo — APPROVED** [review]

**Pull request merged**